File: CS/Data/Gender.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;

namespace DataAnnotationAttributes.Model {
    public enum Gender { Female = 0, Male = 1 }
    public class GenderList: ReadOnlyCollection<Gender> {
        public static IList<Gender> GetGenderValues() {
            IList res = Enum.GetValues(typeof(Gender));
            res = (IList)res.Cast<Gender>();
            return (IList<Gender>)res;
        }
        public GenderList()
            : base(GetGenderValues()) {
        }
    }
}

File: VB/Data/Gender.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Linq

Namespace DataAnnotationAttributes.Model
	Public Enum Gender
		Female = 0
		Male = 1
	End Enum
	Public Class GenderList
		Inherits ReadOnlyCollection(Of Gender)
		Public Shared Function GetGenderValues() As IList(Of Gender)
			Dim res As IList = System.Enum.GetValues(GetType(Gender))
			res = CType(res.Cast(Of Gender)(), IList)
			Return CType(res, IList(Of Gender))
		End Function
		Public Sub New()
			MyBase.New(GetGenderValues())
		End Sub
	End Class
End Namespace